src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -2482,18 +2482,23 @@ await modelResource.ProcessContainerRuntimeArgValues(
             {
                 // If overriding the default resource CA bundle, then we want to copy our bundle to the well-known locations
                 // used by common Linux distributions to make it easier to ensure applications pick it up.
-                foreach (var bundlePath in bundlePaths!)
+                // Group by common directory to avoid creating multiple file system entries for the same root directory.
+                foreach (var bundlePath in bundlePaths!.Select(bp =>
+                {
+                    var filename = Path.GetFileName(bp);
+                    var dir = bp.Substring(0, bp.Length - filename.Length);
+                    return (dir, filename);
```

Comment:
Do we need to be more defensive here?

Reply:
Only reason this would result in invalid directory or filename is if the user customized the paths and then it would be DCP that would actually return an error due to invalid create file arguments (missing filename or invalid path).